extensions/dacpac/src/wizard/pages/deployConfigPage.ts

```diff
@@ -190,13 +196,29 @@ export class DeployConfigPage extends DacFxConfigPage {
 
 		this.updateNewRadioButton();
 
-		return this.displayRadioButtons(this.upgradeRadioButton, this.newRadioButton);
+		return this.createRadioButtonFlexContainer(this.upgradeRadioButton, this.newRadioButton);
+	}
+
+	/*
+	Function that enables the upgrade radio button if databases exist.
+	*/
+	private async enableUpgradeRadioButton(): Promise<azdata.FormComponent> {
+		/*
+		Enable the upgrade radio button and call the updateUpgradeRadioButton function to update the
+		upgrade radio button accordingly.
+		*/
+		this.upgradeRadioButton.enabled = true;
+		this.upgradeRadioButton.checked = true;
+
+		this.updateUpgradeRadioButton();
+
+		return this.createRadioButtonFlexContainer(this.upgradeRadioButton, this.newRadioButton);
```

Comment:
I believe the enableUpgradeRadioButton function should be an async function as it is waiting for the list of databases to be loaded in the dropdown if they exist, and if it isn't an async function, it is set to new databases radio button by default. The disableUpgradeRadioButton on the other hand, doesn't need to be async so I will change that.

Regarding the radio button flex container function, in a case where first there are no databases and the radio button is defaulted to new databases, and then if we create databases, we have to tell the DOM to update the radio button flex component as well, as just enabling the upgrade button to true will not enable it in the DOM, so I think we need to call the radio button flex container function in the enableUpgradeRadioButton function.

Reply:
if a function is async, there should be an await in the function and there isn't an await in enableUpgradeRadioButton(). 

The DOM only gets updated when the formBuilder is updated (like on line 239 when the database dropdown is removed). enableUpgradeRadioButton() is returning the flexContainer formComponent, but nothing is being done with it where enableUpgradeRadioButton() is called on line 172.

Reply:
I checked out these changes and realized why the upgrade existing radio button wasn't getting enabled - populateDeployDatabaseDropdown() isn't getting called when the server dropdown changes. I think this should be fixed by adding a listener to serverDropdown.onValueChanged() to call populateDeployDatabaseDropdown(), like https://github.com/microsoft/azuredatastudio/blob/c963e6ec081efd380a0dbdd92530b323960f3562/extensions/dacpac/src/wizard/api/dacFxConfigPage.ts#L43

Reply:
or another option is to override the populateDatabaseDropdown() function that's inherited from dacFxConfigPage. I don't think there's a reason why they should both be getting called

Reply:
> if a function is async, there should be an await in the function and there isn't an await in enableUpgradeRadioButton().
> 
> The DOM only gets updated when the formBuilder is updated (like on line 239 when the database dropdown is removed). enableUpgradeRadioButton() is returning the flexContainer formComponent, but nothing is being done with it where enableUpgradeRadioButton() is called on line 172.

Okay, thanks for the clarification, I will change it accordingly.



> I checked out these changes and realized why the upgrade existing radio button wasn't getting enabled - populateDeployDatabaseDropdown() isn't getting called when the server dropdown changes. I think this should be fixed by adding a listener to serverDropdown.onValueChanged() to call populateDeployDatabaseDropdown(), like
> 
> https://github.com/microsoft/azuredatastudio/blob/c963e6ec081efd380a0dbdd92530b323960f3562/extensions/dacpac/src/wizard/api/dacFxConfigPage.ts#L43

I feel a bit lost here, when I switched to a server with databases, it populates the dropdown but it throws errors in the console log and also throws an error in step 3 in the application wizard, could you suggest what to do here?

Reply:
@kisantia Quick question, how do you call the populateDeployDatabaseDropdown function from the serverDropdown function? It is two seperate files, and I'm not sure how to export it to the dacFxConfigPage.tx file.

Reply:
in deployConfigPage.ts, you can add the listener in start(), but it's better to just rename populateDeployDatabaseDropdown to populateDatabaseDropdown so that the one in dacFxConfigPage.ts doesn't get called

Reply:
can you share the errors?

Reply:
I followed your advice and changed the populateDeployDatabaseDropdown function in deployConfigPage to populateDatabaseDropdown, and made an additional check to see if the selected server in the dropdown has databases or doesn't. When I switch to a server with databases, the database component never stops loading. 
![errorscreen](https://user-images.githubusercontent.com/29865459/97505360-081d5300-1936-11eb-85b3-afcab247f50a.png)

However,  when I toggle between the new radio button and the upgrade radio button, it loads the databases.

![errorscreen2](https://user-images.githubusercontent.com/29865459/97505373-0c497080-1936-11eb-8d08-67d2697aa5c3.png)

Reply:
Can you show the error in the console in the developer tools?

When I was playing around with this yesterday, I was seeing an error something like the name of the databaseDropdown.value couldn't be resolved. This error can be fixed by moving updating the dropdown's values (line 180) to before the disable/enable upgradeExisting radio button check

Reply:
Yes, I have currently just commented out the lines causing the console error, but I will try this out now.

![consoleerror1](https://user-images.githubusercontent.com/29865459/97513640-0cec0200-194a-11eb-828a-ca8833cd453a.png)